_posts/en/resources/2019-09-20-electronic-ultrasonic-sensor-arduino.md

```diff
@@ -0,0 +1,184 @@
+
+
+#### HC-SR04 Timing diagram
+
+* The circuit needs to supply a short 10uS pulse to the trigger input to start the ranging, and then the module will send out
+an 8 cycle burst of ultrasound at 40 kHz and raise its echo. 
+* The Echo is avdistance object that is pulse width and the range in proportion .You can
+calculate the range through the time interval between sending trigger signal and
+receiving echo signal. Formula: uS / 58 = centimeters or uS / 148 =inch; or: the
+range = high level time * velocity (340M/S) / 2; we suggest to use over 60ms
+measurement cycle, in order to prevent trigger signal to the echo signal. 
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-timing-diagram.png' | relative_url}}"/>
+
+
+
+### Component diagram
+
+The following shows the a breadboard component diagram which will be used for this project.
+
+<img class="img-fluid" src="{{'assets/posts/2019-09-20-arduino-ultrasonic-sensor/ultrasonic-range-sensor.png' | relative_url}}"/>
+
+
+### Code
+
+* The following code snippet can be used  (note it's partially incomplete - you can complete yourself)
+
+```
+
+int buzzer = 3, ledRed = 6, ledYellow = 9, ledGreeen = 10;
+int triggerPin = 4;
+int echoPin = 2;
+
+long readTime(int triggerPin, int echoPin){
+  
+  // In order for the sensor to send out a pulse the trigger needs to be set low for 2us
+  // Then set HIGH for 10us
+     
+  pinMode(triggerPin, OUTPUT);
+  digitalWrite(triggerPin, LOW); 
+  delayMicroseconds(2); 
+
+  digitalWrite(triggerPin, HIGH); 
+  delayMicroseconds(10); 
+  digitalWrite(triggerPin, LOW); 
+  pinMode(echoPin, INPUT); 
+  return pulseIn(echoPin, HIGH);
+}
+
+void setup() {
+  Serial.begin(9600);
+  pinMode(buzzer, OUTPUT); 
+  pinMode(ledRed,OUTPUT); 
+  pinMode(ledYellow,OUTPUT); 
+  pinMode(ledGreeen,OUTPUT);
+}
+
+void loop() {
+  int distance; 
+  distance = 0.01715 * readTime(triggerPin, echoPin); // From calculation above Distance = 0.0343 / 2 * readTime
+  Serial.print("distance: "); 
+  Serial.print(distance);
+  Serial.println("cm"); 
+  
+  if(distance > 100){
+    	green(); 
+    }
+  
+  if(distance > 70 && distance < 100){
+    	yellow(); 
+    }
+  
+  if(distance < 70){
+    	red(); 
+    }
+  
+}
+
+void red(){
+
+    // light the red
+    // sound the buzzer 
+    // delay 100ms
+}
+
+void green(){
+
+   digitalWrite(ledGreeen, HIGH);
+   digitalWrite(ledYellow, LOW); 
+   digitalWrite(ledRed, LOW); 
+   digitalWrite(buzzer, LOW);
+   delay(100);
+}
+
+void yellow(){
+  // light yellow..
+}
+
+```
+
+
+### Resources
+* Lookup the data sheet for the HC-SR04 to find more out about its specifications and how to operate.
+* Prior to building the circuit for real, you could use [Tinkercad](http://tinkercad.com) to prototype
```

Comment:
*you could use*, I would put a stronger statement like: *We strongly recommend...*